crates/hir_ty/src/lower.rs

```diff
@@ -1101,9 +1101,13 @@ fn type_for_type_alias(db: &dyn HirDatabase, t: TypeAliasId) -> Binders<Ty> {
     let resolver = t.resolver(db.upcast());
     let ctx =
         TyLoweringContext::new(db, &resolver).with_type_param_mode(TypeParamLoweringMode::Variable);
-    let type_ref = &db.type_alias_data(t).type_ref;
     let substs = Substs::bound_vars(&generics, DebruijnIndex::INNERMOST);
-    let inner = Ty::from_hir(&ctx, type_ref.as_ref().unwrap_or(&TypeRef::Error));
+    let inner = if db.type_alias_data(t).is_extern {
+        Ty::simple(TypeCtor::ForeignType(t))
```

Comment:
this would blow up (later) if the type alias has generics; even if that's not actually currently possible, I think the solution is just to do `Ty::apply(TypeCtor::ForeignType(t), substs)`, so that would be better

Reply:
Ok, will do!

Reply:
Done~